### PR TITLE
[e2e] Refactor code

### PR DESCRIPTION
**Pull request by demonCoder95**
Refactor code to address linter comments

1. Refactor deprecated `wait.Poll*` functions.
2. Refactor deprecated `io/ioutil` package.

### CODE REVIEW

**Comment by demonCoder95** [comment]
👍

File: test/e2e/util.go [hunk 1]
```diff
@@ -52,11 +52,11 @@ var (
 	pollLongTimeout = 5 * time.Minute
 )
 
-// type ConditionFunc func() (done bool, err error)
-// Poll(interval, timeout time.Duration, condition ConditionFunc)
+// type wait.ConditionWithContextFunc func(context.Context) (done bool, err error)
+// PollUntilContextTimeout(ctx context.Context, interval, timeout time.Duration, immediate bool, condition wait.ConditionWithContextFunc)
```
Comment (linki) [inline test/e2e/util.go]: These two comment lines seem a little misplaced. Can we just remove them?

Comment (demonCoder95) [reply test/e2e/util.go]: Sure

Comment (demonCoder95) [reply test/e2e/util.go]: Done

**Comment by demonCoder95** [comment]
👍

**Comment by linki** [comment]
👍

**Pull request merged**